cfn-resources/cfn-publishing-automation/README.md

```diff
@@ -0,0 +1,43 @@
+# CFN publishing automation document for AWS System Manager
+
+** *Documents in this directory intended for internal use only by MongoDB, Inc.*
+
+cfn-3P-provider-register.yaml file contains the automation document that is used to publish
+MongoDB's CFN resources to AWS public registry.
+
+This automation document builds and registers 3rd party resource providers. This automation document
+can register resources in a single AWS account or in multiple accounts and regions. However,
+multi-account and regions require a role. Please refer to
+the [SSM multi-account ](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-automation-multiple-accounts-and-regions.html)
+documentation.
+
+This document builds and registers 3rd party resource providers build using the following plugins:
+
+* JAVA [link to another webpage](https://aws.amazon.com/)
+* Go [link to another webpage](https://aws.amazon.com/)
+* Python (Coming soon) [link to another webpage](https://aws.amazon.com/)
+
+## Making updates to the automation:
+
+- The .yaml script is a copy of "CFN-MongoDB-Atlas-Resource-Register" automation document that can
+  be found in AWS Systems Manager (please reach out to team for access)
+
+To update the script in AWS you will need to create a new version of the document. Please follow
+below instructions:
+
+1. Create a new test document in AWS account with your updated script and test your changes.
+2. Upon successful testing, raise a PR with your changes in this repo.
+3. Go to AWS publishing account -> Systems Manager -> Shared Resources in left menu -> Documents.
+   Then locate `CFN-MongoDB-Atlas-Resource-Register` document under
+   and notice the current latest version of the document in the dropdown.
+4. On PR approval please ensure to use the following format for commit message when _(squash &)_
+   merging your PR:
+   `AWS automation update: v<newVersion#> [<brief message about changes>]`.
+   - For example, `AWS automation update: v52 [fix build status in step 6]`
```

Comment:
[q] I really like the version approach here, but it's quite manual and will I can imagine it will easily fall out of sync.

Have you checked is there any way to source the script from a git file or link? This would be common for IaC approaches - curious why AWS doesn't support it.

Reply:
Looks like they might have a python or ansible support: https://docs.aws.amazon.com/systems-manager/latest/userguide/integration-remote-scripts.html

This isn't blocking the PR, more a future improvement, but if we're going to use Evergreen in the future, probably overkill.

Reply:
I think there are ways to use automation document from an S3 bucket but not from a git repo/link. Another way would be to use AWS SDK for Systems Manager to create and run a automation from our machine, probably then we'll be able to maintain the code in our repo (needs more investigation)

Reply:
We could try this https://docs.aws.amazon.com/systems-manager/latest/userguide/run-remote-documents.html

Reply:
sgtm, something to consider when we move to Evergreen